File: Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram.java
class Solution {
    public int minSteps(String s, String t) {
        Map<Character, Integer> sCount = new HashMap<>();

        for (char c : s.toCharArray()) {
            sCount.put(c, 1 + sCount.getOrDefault(c, 0));
        }

        int change = 0;
        for (char c : t.toCharArray()) {
            if (!sCount.containsKey(c) || sCount.get(c) == 0) {
                change += 1;
            } else {
                sCount.put(c, sCount.get(c) - 1);
            }
        }

        return change;        
    }
}

File: Minimum_Number_of_Steps_to_Make_Two_Strings_Anagram.py
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        s_count = {}

        for c in s:
            s_count[c] = 1 + s_count.get(c, 0)
        
        change = 0
        for c in t:
            if c not in s_count or s_count[c] == 0:
                change += 1
            else:
                s_count[c] -= 1
        
        return change
